File: 141-linked-list-cycle/linked-list-cycle.java

public class Solution {
    public boolean hasCycle(ListNode head) {
        HashSet<ListNode> a=new HashSet<>();
        ListNode temp=head;
        while(temp!=null)
        {
            if(a.contains(temp))
            return true;
            a.add(temp);
            temp=temp.next;
        }
        return false;
    }
}

File: 141-linked-list-cycle/linked-list-cycle.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        if(head==None):
            return None
        slow=head
        fast=head.next
        while(slow and fast):
            if(slow==fast):
                return True
            slow=slow.next
            if(fast.next==None):
                return False
            fast=fast.next.next
        return False
        